### PR TITLE
feat: complete frontend analytics attribution

### PR DESCRIPTION
**Pull request by jwalkingjew**
## Summary

- add success-level analytics for newsletter signup, profile publication, and comment edits
- preserve the existing centralized comment-create, entity-vote, claim-response, and debate-winner vote paths while adding entity and surface context
- attribute person-profile navigation from claim response lists, debate media, claim provenance, comment authors, and the debate people hub
- give email and account forms, profile editing, debate hub controls, and debate side-panel actions stable analytics types, labels, and intents
- retain claim and debate context when comments open through the shared global panel
- keep shared controls reusable without leaking hub attribution into the standalone rematch flow or putting live counts into analytics labels
- keep payloads privacy-safe: no newsletter email, OTP, comment text, or edited profile values are recorded

## Audit results

| User action | Coverage |
| --- | --- |
| Newsletter popup | Form started and submitted plus `signup_completed` only after the API succeeds |
| Account creation from popup | Typed account form plus existing Privy `signed_up` / `signed_in`, attributed to `explore_email_capture` |
| Comment create/reply | Centralized `comment_created` after successful publish, with entity type, surface, space, and parent ID |
| Comment edit | Centralized `content_edited` after successful publish |
| Shared comments panel | Claim and debate types survive the panel handoff, including Explore cards |
| Entity and claim voting | Existing shared `vote_cast` path covers curation, stance, veracity, clearing and switching, and indexed outcomes |
| Debate winner voting | Existing `vote_cast` operation path covers submitted and indexed outcomes |
| Profile editing | `published_edit` only after a real successful profile publication; no-op saves are excluded |
| Person navigation | `graph_relationship_followed` with personal-space or person IDs and originating surface |
| Debate hub and panels | Stable auto-capture labels and intents for entry points, tabs, availability, hub actions, comments, claims, share, and closes |

## Validation

- `bun run test` before the Copilot-review fixes: 572 files, 6,840 tests passed
- full suite after the fixes: 6,839/6,842 tests passed; the three unrelated high-load failures each passed immediately in isolation
- all affected test files after rebasing onto latest `master`: 10 files, 252 tests passed
- both new regression tests were mutation-checked by removing their fix and confirming they failed
- `bun run typecheck` in `apps/web`
- ESLint on changed analytics files
- `git diff --check`

### CODE REVIEW

File: apps/web/core/debates/browse/use-open-debater-profile.ts [hunk 1]
```diff
@@ -20,14 +21,21 @@ import { useSpace } from '~/core/hooks/use-space';
...
       // The video behind is one large play/pause button.
       event.stopPropagation();
-      if (participant && profileEntityId) openSidePanel(profileEntityId, participant.profile_space_id, false);
+      if (participant && profileEntityId) {
+        personProfileOpened(participant.profile_space_id, resolvedProfileEntityId, {
+          interaction_surface: 'debate_media',
+          navigation_mode: 'entity_side_panel',
+        });
+        openSidePanel(profileEntityId, participant.profile_space_id, false);
+      }
```
[the remark this answers is not in the document]
Comment (jwalkingjew) [reply apps/web/core/debates/browse/use-open-debater-profile.ts:37]: Copilot is right about the failure mode, but I did not use the suggested one-off reorder. That would protect only this click and leave every other analytics call in this PR able to throw into product code. In 74b37e186 I contained runtime exceptions at the shared analytics invocation boundary, log and drop the failed event so the pending queue cannot stall, and keep the primary action running. The regression test failed against the prior implementation, passed with the fix, and failed again when I deliberately removed the guard.

File: apps/web/core/analytics.ts [hunk 2]
```diff
@@ -270,6 +270,27 @@ export function personalSpaceViewed(personalSpaceId: string, properties: Analyti
...
+  profileSpaceId: string,
+  personEntityId?: string | null,
+  properties: AnalyticsProperties = {}
+) {
+  graphRelationshipFollowed(personEntityId || profileSpaceId, {
+    source: 'person_profile',
+    graph_entity_type: personEntityId ? 'person' : 'personal_space',
+    profile_space_id: profileSpaceId,
+    ...properties,
+  });
```
[the remark this answers is not in the document]
Comment (jwalkingjew) [reply apps/web/core/analytics.ts]: Copilot is right. The profile adapter can map a null Person entity ID to the personal-space ID while still producing a valid profile link, so truthiness alone misclassified that fallback. In 74b37e186 the shared personProfileOpened helper treats missing or equal IDs as personal_space, covering every caller added by this PR. The regression test failed against the prior implementation, passed with the fix, and failed again when I deliberately restored the old classification.

File: apps/web/core/analytics.ts [hunk 3]
```diff
@@ -270,6 +270,29 @@ export function personalSpaceViewed(personalSpaceId: string, properties: Analyti
...
+}
+
+export function personProfileOpened(
+  profileSpaceId: string,
+  personEntityId?: string | null,
+  properties: AnalyticsProperties = {}
+) {
+  const resolvedPersonEntityId = personEntityId && personEntityId !== profileSpaceId ? personEntityId : null;
```
[the remark this answers is not in the document]
Comment (jwalkingjew) [reply apps/web/core/analytics.ts]: You're right about the problem. I fixed this with the repository's existing ID.equals comparator rather than adding another normalization path, so dashed/case variants of the same UUID stay classified as the personal space. The regression covers both variants; restoring the strict string comparison makes it fail.

File: apps/web/partials/entity-page/entity-vote-buttons.tsx [hunk 4]
```diff
@@ -609,7 +609,14 @@ function VoterRow({ profile }: { profile: Profile }) {
   );
 
   if (profile.profileLink) {
-    return <Link href={profile.profileLink}>{content}</Link>;
+    return (
+      <Link
+        href={profile.profileLink}
+        onClick={() => personProfileOpened(profile.spaceId, profile.id, { interaction_surface: 'claim_vote_list' })}
```
[the remark this answers is not in the document]
Comment (jwalkingjew) [reply apps/web/partials/entity-page/entity-vote-buttons.tsx]: You're right. I enumerated this through the shared responder-row path: stance/veracity navigation retains claim_vote_list, while curation on comments, profiles, topics, debates, and other entities now uses entity_vote_list. The regression exercises both surfaces; restoring the hard-coded claim surface makes the curation case fail.

File: apps/web/partials/explore/email-capture-account-step.tsx [hunk 5]
```diff
@@ -41,7 +41,13 @@ export function AccountStep({ email, onGiveUp }: { email: string; onGiveUp: () =
...
     state: otpState,
   } = useLoginWithEmail({
-    onComplete: args => trackPrivyAuth(args, { auth_flow: 'manual_login', link_source: 'explore_email_capture' }),
+    onComplete: args =>
+      trackPrivyAuth(args, {
+        auth_flow: 'manual_login',
+        link_source: 'explore_email_capture',
+        form_type: 'account',
+        signup_surface: 'explore_email_capture',
+      }),
```
[the remark this answers is not in the document]
Comment (jwalkingjew) [reply apps/web/partials/explore/email-capture-account-step.tsx:50]: You're right, and the old comment's navbar fallback assumption was stale because that tracker is armed only by its own button. The failed-code fallback now keeps AccountStep mounted (the parent hides it behind Privy), gives usePrivySignIn the Explore attribution, and closes the attempt from a new dismissal/error callback. The visible sequence remains popup → Privy modal → closed. Separate regressions fail if the old early unmount or the missing dismissal callback is restored.

File: apps/web/core/debates/matchmaking/hub-pill-button.tsx [hunk 6]
```diff
@@ -21,14 +25,23 @@ export function HubPillButton({
...
+  const resolvedAnalyticsLabel =
+    analyticsLabel ??
+    rest['aria-label'] ??
+    (typeof children === 'string' ? `Debate hub ${children}` : 'Debate hub action');
+
   return (
     <button
       type="button"
+      data-geo-analytics-label={resolvedAnalyticsLabel}
+      data-geo-analytics-intent={analyticsIntent}
```
[the remark this answers is not in the document]
Comment (jwalkingjew) [reply apps/web/core/debates/matchmaking/hub-pill-button.tsx]: You're right. I enumerated the same class across the hub rather than patching only this component: cancel/withdraw, request-overflow trigger and actions, status/space/topic filter triggers/options/clear actions, filter switches, and schedule controls now have stable labels/intents through one shared helper. Filter metadata is opt-in so feed/profile/rematch consumers of the shared menus are not mislabeled as hub activity. Component/integration regressions cover the reported control families; mutating the shared metadata helper makes them fail.

File: apps/web/core/debates/matchmaking/claims-tab.tsx [hunk 7]
```diff
@@ -1306,6 +1306,7 @@ export function SpaceTopicFilters({
       {leading}
       <HubMultiFilterMenu
         label={spaceMenuLabel}
+        analyticsName="Space"
```
[the remark this answers is not in the document]
Comment (jwalkingjew) [reply apps/web/core/debates/matchmaking/claims-tab.tsx]: Valid. I enumerated every shared filter-menu use in this PR rather than changing only SpaceTopicFilters. Shared debate filters now receive an explicit owning surface, and the filter name and surface travel together so partial analytics configuration is not possible. Hub Space, Topic, and Status controls retain hub attribution; rematch Space, Topic, and Claims source controls now use Debate rematch labels with filter_debate_rematch. I left the feed and profile consumers unattributed because they are outside the debate surfaces and changing their analytics taxonomy here would be out of scope. The regression test failed with Debate hub Space filter when I deliberately forced this rematch call back to hub, and passes with the fix.

File: apps/web/core/debates/matchmaking/filter-switch.tsx [hunk 8]
```diff
@@ -29,6 +31,7 @@ export function FilterSwitch({
       role="switch"
       aria-checked={checked}
       aria-label={label}
+      {...hubAnalyticsAttributes(label, 'filter_debates_hub')}
```
[the remark this answers is not in the document]
Comment (jwalkingjew) [reply apps/web/core/debates/matchmaking/filter-switch.tsx]: Valid. I enumerated both shared switch wrappers and every caller. FilterSwitch, MatchesOnlySwitch, and HideMyPositionsSwitch now require an owning analytics surface; hub callers pass hub and rematch callers pass rematch. I chose a required surface rather than an optional opt-in because a default is what allowed the attribution leak and would make the same regression easy for the next shared caller. No visual or filtering behavior changed. The rematch switch test failed with Debate hub Matches only and filter_debates_hub when I deliberately restored the old hard-coded hub path, and passes with the fix.

File: apps/web/core/debates/matchmaking/hub-pill-button.tsx [hunk 9]
```diff
@@ -21,14 +25,23 @@ export function HubPillButton({
   variant = 'secondary',
   pendingLabel,
   pending = false,
+  analyticsLabel,
+  analyticsIntent = 'debates_hub_action',
```
[the remark this answers is not in the document]
Comment (jwalkingjew) [reply apps/web/core/debates/matchmaking/hub-pill-button.tsx]: Valid. I traced every action rendered by HubQueryState, not just the empty-state line: sign-in, account-warmup retry, ordinary retry, and empty-state actions now all receive the caller owning analytics surface. The rematch page passes rematch; all hub callers pass hub. HubPillButton still defaults to hub for direct hub-only uses, while shared state callers must be explicit, preserving existing specific label and intent overrides. The empty-state regression test failed with Debate hub Clear filters when I deliberately restored the old default on that path, and passes with Debate rematch plus debate_rematch_action after the fix.

File: apps/web/core/hooks/use-privy-sign-in.ts [hunk 10]
```diff
@@ -77,6 +79,7 @@ export function usePrivySignIn(onComplete?: () => void, options?: UsePrivySignIn
     onError: () => {
       requestedRef.current = false;
       requestedAnalyticsRef.current = undefined;
+      optionsRef.current?.onError?.();
     },
```
[the remark this answers is not in the document]
Comment (jwalkingjew) [reply apps/web/core/hooks/use-privy-sign-in.ts:84]: Valid. Privy error callbacks are global to the hook integration, so this hook must ignore an error unless it armed the attempt, matching the existing completion guard. I added that ownership check before clearing state or invoking the caller callback. I did not suppress errors for attempts this hook actually starts; those still clear the attempt and notify the email popup. The only user-visible edge-case change is that an unrelated Privy failure or dismissal no longer closes an in-progress email signup popup. Removing the new guard made the new test fail while the other seven hook tests passed; restoring it makes all eight pass.

**Comment by jwalkingjew** [comment]
Addressed Copilot’s previously missed summary finding from the latest review. The finding was valid: the People-tab source comment still said the person link had no click handler after this PR added an observational analytics handler. I enumerated the same stale statement in the PR code and corrected both the component comment and the corresponding test-suite comment in 8a72c16e5. I did not change the handler or navigation behavior: the existing test continues to verify that the handler records analytics while Next owns navigation. I also did not add a source-text or mutation test for a comment-only correction, because that would test prose rather than product behavior and would not follow this repository’s testing patterns. There was no inline thread attached to this summary-only finding, so this PR comment records the decision and resolution.

**Comment by jwalkingjew** [comment]
Addressed all three previously missed findings from Copilot’s latest summary in c73122404. Copilot did not create inline threads for these items, so this comment records each decision and resolution.

1. Keep looking attribution: valid. The empty and non-empty controls invoke the same rematch action but had different labels. The non-empty button now uses the shared rematch surface, producing Debate rematch Keep looking and debate_rematch_action just like the empty-state path. I enumerated production rematch action overrides; no other one-off production label remains. Forcing this call back to the hub surface makes the new regression fail.

2. Request overflow surface: valid misattribution, but the suggested implication that the intents were hub-specific was not correct. open_debate_request_options and manage_debate_request describe the actions without naming a surface, so renaming them would split an established event family without benefit. I kept those intents and added a shared debate-action surface helper. The hub card now explicitly uses the hub surface and the app-wide popup explicitly uses the request-popup surface, so their stable labels are distinct. Forcing the popup back to hub makes the new trigger/action regression fail.

3. Pending comment author ID: valid. Rather than patching only one comment link, personProfileOpened now rejects the repository’s canonical pending personal-space sentinel at the shared analytics boundary. This covers all three comment-author links and any other current or future profile-open caller without changing navigation to the intentionally navigable pending page. Removing the guard makes the new analytics regression emit graph_relationship_followed with pending:0x123 and fail.

No visual, navigation, or mutation behavior changed, and no out-of-scope issues were found. Final focused result: 235 tests pass; type-check, lint, formatting, and diff checks pass.

File: apps/web/core/analytics.ts [hunk 11]
```diff
@@ -270,6 +273,31 @@ export function personalSpaceViewed(personalSpaceId: string, properties: Analyti
...
+  profileSpaceId: string,
+  personEntityId?: string | null,
+  properties: AnalyticsProperties = {}
+) {
+  if (isPendingPersonalSpaceId(profileSpaceId)) return;
+
+  const resolvedPersonEntityId = personEntityId && !ID.equals(personEntityId, profileSpaceId) ? personEntityId : null;
+
+  graphRelationshipFollowed(resolvedPersonEntityId ?? profileSpaceId, {
+    source: 'person_profile',
```
[the remark this answers is not in the document]
Comment (jwalkingjew) [reply apps/web/core/analytics.ts:294]: I checked the premise against the bundled analytics runtime and am not changing this. `registry.sources` is not enforced as an allowlist for the browser event `source` property: the shipped bundle's own helpers emit `signup_form`, `publish_flow`, `commenting`, `mode_toggle`, `content_edit`, and other values absent from that array, while `validateEvent` checks required fields and measurement contracts but never compares `source` to `registry.sources`. `capture` forwards these values unchanged.

I also enumerated the same class across this PR rather than stopping at the two cited lines. Its newly introduced source values are `person_profile`, `profile_editor`, and `signup_form`; the last is itself an official runtime-helper value, and the existing wrapper/runtime already use similarly event-specific sources. Replacing `person_profile` with `graph` and `profile_editor` with a broader value would discard producer context without preventing an actual validation failure. `graph_entity_type` and `interaction_surface` are complementary dimensions, not evidence that `source` must be one of the registry metadata entries.

No code or test changed. A regression test enforcing membership in `registry.sources` would codify a constraint the runtime does not implement and would fail against its own supported helpers.

**Review by ohohoreilly — COMMENTED** [review]
Read every non-test hunk across the 66 files. **No correctness issues found.** The riskier spots all hold up:

- **`use-edit-profile.ts`:** `settleSuccess(recordPublishedEdit = true)` has only two callers. The no-op path passes `false` and the `makeProposal` `onSuccess` calls it bare, so no callback leaks an unexpected argument into it. The only other `published_edit` sender is `review-changes.tsx`, so a profile publish isn't counted twice.
- **Email-capture sign-in fallback:** the card stays mounted behind Privy's modal (hidden, because `isModalOpen` counts as an overlay while `wantsAccount` is true). Dismissing the modal reaches `giveUpRef.current()` through the new guarded `onError` in `usePrivySignIn`. The navbar's `PrivyConnectButton` now only tracks logins it started, so there's no double-counted sign-in.
- **`usePrivySignIn` `onError`:** the early return behaves the same as the old unconditional clear, and the caller's callback only fires for a sign-in this hook started.
- **`editComment` wrapper:** keeps the underlying `Promise<boolean>` and records the edit only when the save succeeds.
- **Wiring:**
  - `content_edited`, `signup_completed` and `graph_relationship_followed` are registered in `public/geo-analytics-*.js`.
  - `SmallButton`, `Button` and `PrefetchLink` pass `data-*` and `onClick` through.
  - The rematch page passes `analyticsSurface="rematch"` to every shared control, so no hub labels leak into it.

**One optional note, not a bug:** the generic `entity-page-body.tsx:121` still renders `<CommentSection>` without `targetEntityType`, so comments posted on a debate's full entity page get tagged `entity`, not `debate`. The PR only promises debate and claim context through the shared panel, so this may be fine as is. Worth a one-line prop if you want debate comments attributed wherever they're posted.

**Comment by jwalkingjew** [comment]
Thanks for the thorough review. I confirmed the main conclusion: I did not find a correctness issue in the paths you called out, and there were no inline threads to address.

I did take the optional comment-attribution note. The normal debate browse view already passes `targetEntityType="debate"`, but the generic footer is also the edit/fallback and side-panel path, where a Debate was still falling back to `entity`. In `68dce2bc9`, the footer now derives the logical comment target from the entity data it already reads. I checked the same class rather than patching only Debate: Claim and Topic custom comment views retain their explicit types, and the shared derivation recognizes Claim, Topic, and Debate whenever one reaches the generic footer.

The regression was written first and failed against the previous behavior. I also deliberately replaced the derivation with the old `entity` fallback and confirmed the test failed before restoring the fix. This changes only analytics payload context; it does not change rendering, navigation, or publishing.

Rebasing across current `master` exposed one related new caller that landed after this PR's prior base: the profile Debates tab now uses the shared `FilterSwitch`, whose owning analytics surface this PR intentionally made required. In `a02dcfcfe`, that caller now uses a distinct `profile-debates` surface (`Profile debates Show hidden` / `filter_profile_debates`) rather than being mislabeled as hub or rematch activity. Its regression also failed before the new surface existed.

Validation after the rebase: 601 broad focused tests and 25 post-fix tests pass; typecheck, lint, formatting, and diff checks pass. No device-visible behavior changed.
